### PR TITLE
Alerting: Memoize labels suggestions calculation

### PR DESCRIPTION
**Pull request by konrad147**
**What is this feature?**
Performance optimization of Alert Rule Form

**Special notes for your reviewer:**
The `useGetCustomLabels` hook gets fired on each keystroke (because it's parent component uses the form context and rerenders on each input change)
This PR adds a memo around calculating the suggested labels to alleviate the pain of editing other form fields

Users with a large number of labels and keys suffered from performance issues

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by soniaAguilarPeiron — APPROVED** [review]
LGTM! good catch!

**Review by VikaCep — APPROVED** [review]

**Review by gillesdemey — APPROVED** [review]
LGTM just a few nits

File: public/app/features/alerting/unified/components/rule-editor/LabelsField.tsx [hunk 1]
```diff
@@ -38,29 +38,35 @@ const useGetCustomLabels = (dataSourceName: string): { loading: boolean; labelsB
...
+        )
+      )
+    );
 
-  labels.forEach((label: Record<string, string>) => {
-    Object.entries(label).forEach(([key, value]) => {
-      labelsByKey[key] = [...new Set([...(labelsByKey[key] || []), value])];
+    labels.forEach((label: Record<string, string>) => {
+      Object.entries(label).forEach(([key, value]) => {
+        labelsByKey[key] = [...new Set([...(labelsByKey[key] || []), value])];
```
[the remark this answers is not in the document]
Comment (konrad147) [reply public/app/features/alerting/unified/components/rule-editor/LabelsField.tsx]: I changed the way we get the labels and their values. I hope it's easier to understand now

**Pull request merged**